src/storage-blob-preview/azext_storage_blob_preview/_client_factory.py

```diff
@@ -76,6 +78,18 @@ def cf_blob_service(cli_ctx, kwargs):
 
 
 def cf_blob_client(cli_ctx, kwargs):
+    if kwargs.get('blob_url'):
+        t_blob_client = get_sdk(cli_ctx, CUSTOM_DATA_STORAGE_BLOB, '_blob_client#BlobClient')
+        credential = get_credential(kwargs)
+        # del unused kwargs
+        kwargs.pop('connection_string')
+        kwargs.pop('account_name')
+        kwargs.pop('container_name')
+        kwargs.pop('blob_name')
+        return t_blob_client.from_blob_url(blob_url=kwargs.pop('blob_url'),
+                                           credential=credential,
+                                           snapshot=kwargs.pop('snapshot', None))
+    kwargs.pop('blob_url')
```

Comment:
It seems `blob_url` can be either `None` or `''` here. Why should we pop it out?

Reply:
Actually if blob url is none or '', it would be invalid. added check for blob url and blob name.